azurecontainerapps.ts

```diff
@@ -0,0 +1,516 @@
+import * as core from '@actions/core';
```

Comment:
From what we chatted about offline: it would be great if we could refactor the code in such a way that we could easily migrate code between the GitHub Action and the Azure DevOps Task since they'll both be written in TypeScript. To do this, we need to consolidate all of the GitHub Action specific code (_i.e._, code that leverages GitHub Action packages on npm or depends on GitHub Action values) to its own helper class for this tool, and we'll have an equivalent in the Azure DevOps Task down the road. This allows the streamlined flow to be copied between the two tools, and any changes we make for one tool doesn't have to be hand-written in the other to account for how it was implemented. This ensures that we have consistent functionality across both tools and don't need to worry about tracking what is and isn't supported.

From scanning over this PR, it looks like the following components are GitHub Actions specific and would need to be ported over into a new `GitHubActionsToolHelper` file:

```
@actions/core
=============
core.getInput
core.setFailed

core.debug
core.error
core.info
core.warning


@actions/io
===========
io.which


@actions/exec
=============
exec.exec
exec.ExecOptions (I think this is used with the above method call)


Miscellaneous
=============
process.env.GITHUB_RUN_ID
process.env.GITHUB_RUN_NUMBER (we just need equivalent environment variables in ADO for these two)

getContainerAppName()
  - includes GitHub Action in naming schema (just need a function for generating the default ACA name)

Telemetry arguments
  - CALLER_ID=github-actions-v1
  - --event-name ContainerAppsGitHubActionV1 (just need a function for fetching telemetry values)
```

Reply:
These changes have been made.